server/src/main/java/org/schemarepo/server/RESTRepository.java

```diff
@@ -82,81 +85,66 @@
+   * @return All subjects in the repository, serialized with {@link org.schemarepo.RepositoryUtil#subjectsToString(Iterable)}
    */
   @GET
   public Response allSubjects(@HeaderParam("Accept") String mediaType) {
-    if (mediaType.equals(MediaType.APPLICATION_JSON)) {
-      return Response.ok(
-              jsonUtil.subjectsToJson(repo.subjects()),
-              MediaType.APPLICATION_JSON).build();
-    } else {
-      return Response.ok(
-              RepositoryUtil.subjectsToString(repo.subjects()),
-              MediaType.TEXT_PLAIN).build();
-
-    }
+    Renderer renderer = getRenderer(mediaType);
+    return Response.ok(renderer.renderSubjects(repo.subjects()), renderer.getMediaType()).build();
   }
 
   /**
    * Returns all schemas in the given subject, serialized with
-   * {@link RepositoryUtil#schemasToString(Iterable)}
+   * {@link org.schemarepo.RepositoryUtil#schemasToString(Iterable)}
    *
    * @param subject
    *          The name of the subject
-   * @return all schemas in the subject. Return a 404 Not Found if there is no
-   *         such subject
+   * @return all schemas in the subject. Return a 404 Not Found if there is no such subject
    */
   @GET
   @Path("{subject}/all")
-  public Response allSchemaEntries(
-          @HeaderParam("Accept") String mediaType,
-          @PathParam("subject") String subject) {
+  public Response allSchemaEntries(@HeaderParam("Accept") String mediaType, @PathParam("subject") String subject) {
     Subject s = repo.lookup(subject);
     if (null == s) {
       throw new NotFoundException(MessageStrings.SUBJECT_DOES_NOT_EXIST_ERROR);
     }
-    if (mediaType.equals(MediaType.APPLICATION_JSON)) {
-      return Response.ok(
-              jsonUtil.schemasToJson(s.allEntries()),
-              MediaType.APPLICATION_JSON).build();
-    } else {
-      return Response.ok(
-              RepositoryUtil.schemasToString(s.allEntries()),
-              MediaType.TEXT_PLAIN).build();
-
-    }
+    Renderer renderer = getRenderer(mediaType);
+    return Response.ok(renderer.renderSchemas(s.allEntries()), renderer.getMediaType()).build();
   }
 
   @GET
   @Path("{subject}/config")
-  public String subjectConfig(@PathParam("subject") String subject) {
+  public String subjectConfig(@HeaderParam("Accept") String mediaType, @PathParam("subject") String subject) {
     Subject s = repo.lookup(subject);
     if (null == s) {
       throw new NotFoundException(MessageStrings.SUBJECT_DOES_NOT_EXIST_ERROR);
     }
     Properties props = new Properties();
     props.putAll(s.getConfig().asMap());
-    StringWriter writer = new StringWriter();
-    try {
-      props.store(writer, null);
-    } catch (IOException e) {
-      // stringWriter can't throw ... but just in case
-      throw new RuntimeException(e);
-    }
-    return writer.toString();
+    return getRenderer(mediaType).renderProperties(props, "Configuration of subject " + subject);
```

Comment:
The latter syntax involves explicitly setting content type of the response, that's the main (and really the only) difference. (Should there be an exception in processing, it will still result in a 500 status). I'm using the former syntax when the underlying renderer implementations don't actually respond with the appropriately formatted content (e.g. it's not really valid JSON) and thus we don't insist on the response media type and let the client interpret as it wishes. If all the TODOs were implemented, then everything would use the "latter" syntax.